TileFormats/VectorData/README.md

```diff
@@ -1,19 +1,198 @@
 # Vector Data
 
-* [Draft spec](https://github.com/AnalyticalGraphicsInc/3d-tiles/pull/124/files)
-* GitHub issue, [#25](https://github.com/AnalyticalGraphicsInc/3d-tiles/issues/25)
+## Contributors
 
-## Notes
+* Dan Bagnell, [@bagnell](https://github.com/bagnell)
+* Rob Taglang, [@lasalvavida](https://github.com/lasalvavida)
+* Patrick Cozzi, [@pjcozzi](https://twitter.com/pjcozzi)
 
-**Use cases**: Traditional geospatial features: points, polylines, and polygons.  Replacing KML.
+## Overview
 
-**Format**
-* Combination of binary (for positions, normals, etc.) and JSON (for labels, other metadata, etc.).
-* Concise representations for Cesium's [set of geometries](http://cesiumjs.org/2013/11/04/Geometry-and-Appearances/), including extrusions, and billboards and labels.
-   * Need to carefully select the representation for the best trade-off between conciseness and runtime processing.  For example, polygons will likely be pre-triangulated since it only adds indices to the payload, but will be subdivided at runtime since subdivision is fast and increases the vertex payload significantly.
-   * RTC positions for high precision-rendering.
-   * Context-aware compression.
-   * Bounding volume may need to be adjusted at runtime for terrain clamping.
-* Metadata for cracking, morphing, and perhaps label declutter.
+The _Vector_ tile format allows streaming of vector datasets including points, polylines, and polygons.  Points can be represented with a combination of billboard, label, and point graphics primitives.
 
-Could also name this a `Geometry` tile.
+Each point, poyline, and polygon is a _feature_ in the 3D Tiles specification language. 
+
+## Layout
+
+A tile is composed of two sections: a header immediately followed by a body.
+
+**Figure1**: Vector tile layout.
+![layout](figures/layout.jpg)
```

Comment:
Right now, each subbuffer is delta +ZigZag encoded. We can grab the subbuffer, hand it to the webworker where it gets decoded.

If we make it more glTF-like, we could encode the whole buffer instead of each section for polylines, polygons and points. That would add a new step to decode the whole buffer client side. But, what about polygons which don't have heights?

When you quantize a glTF vertex, it doesn't depend on any other vertex. With delta + ZigZag encoding, each value depends on the previous value. So you encode the u, v, and height in separate buffers. Polygon vertices in counter-clockwise order will compress well. Likewise, you want polygons near each other to be consecutive in the buffer for better compression. For polygons, we can index into the u and v buffers no problem. Every other geometry type will have a height value, but how do we index into the height buffer? Fill the polygon heights with empty values? We could make it gzip compress well, but there still might be a lot of empty data when decompressed.